### PR TITLE
Added a verbose option for swarm peers.

### PR DESCRIPTION
**Pull request by csasarak**
This is used like 'ipfs swarm peers -v'. It prints out the normal peers
information, but also the latency of each peer

License: MIT
Signed-off-by: Chris Sasarak <chris.sasarak@gmail.com>

### CODE REVIEW

File: core/commands/swarm.go [hunk 1]
```diff
@@ -50,7 +52,14 @@ var swarmPeersCmd = &cmds.Command{
 'ipfs swarm peers' lists the set of peers this node is connected to.
 `,
 	},
+	Options: []cmds.Option{
+		cmds.BoolOption("verbose", "v",
+			`Also display latency along with peer information in the following form:
+        <peer address> <latency>
```
Comment (Kubuxu) [inline core/commands/swarm.go]: You are using raw string here. Meaning that now this string includes two new lines and also indentation. Please change it to one line.

If you want to split line you can do it like that
```go
"line1" +
"still line1"
```

File: core/commands/swarm.go [hunk 2]
```diff
@@ -50,7 +52,14 @@ var swarmPeersCmd = &cmds.Command{
...
+        `),
```
Comment (Kubuxu) [inline core/commands/swarm.go]: Please add `.Default(false)` it might not change much in this context but clarifies some things.

**Comment by Kubuxu** [comment]
@whyrusleeping might be able to respond to your questions better than I am.

If latency isn't stored or exposed easily then it would be worth pinging in batches of few tenths of peers at a time.

**Comment by lgierth** [comment]
It might be worth to make pinging a separate command, since it doesn't just make the output more verbose, but actually alters the behaviour.

File: core/commands/swarm.go [hunk 3]
```diff
@@ -64,12 +73,32 @@ var swarmPeersCmd = &cmds.Command{
...
 		for i, c := range conns {
 			pid := c.RemotePeer()
 			addr := c.RemoteMultiaddr()
-			addrs[i] = fmt.Sprintf("%s/ipfs/%s", addr, pid.Pretty())
+
+			if verbose {
+				pContext, cancelFn := context.WithTimeout(req.Context(), timeout)
+				defer cancelFn()
+
+				ch, _ := pingService.Ping(pContext, pid)
```
Comment (whyrusleeping) [inline core/commands/swarm.go]: We can grab this information from `n.Peerstore.LatencyEWMA(peer)` no need to ping each peer for it

**Comment by csasarak** [comment]
That's significantly easier than what I was doing, thanks! I'll edit this tonight.

**Comment by csasarak** [comment]
I've incorporated the changes everyone discussed. I failed one of the tests, but haven't merged in master. Do you all prefer that I squash/rebase my commits, I merge master and push to my branch with a merge commit?

**Comment by whyrusleeping** [comment]
@csasarak i can squash the commits here on github. And teamcity always tests the merge commit of your branch and latest master.

**Comment by whyrusleeping** [comment]
This LGTM

**Pull request merged**